src/libmeasurement_kit/dns/getaddrinfo_async.hpp

```diff
@@ -73,18 +73,18 @@ std::vector<Answer> getaddrinfo_async_parse_response(const std::string &name,
         }
         char abuf[128];
         if (inet_ntop(p->ai_family, aptr, abuf, sizeof(abuf)) == nullptr) {
-            throw InetNtopFailureError();
+            return {GenericError{"inet_ntop_failed"}, {}}; // Unexpected
         }
         if (p->ai_family == AF_INET) {
             answer.ipv4 = abuf;
         } else if (p->ai_family == AF_INET6) {
             answer.ipv6 = abuf;
         } else {
-            /* Case excluded above */;
+            throw std::runtime_error("case_excluded_above");
```

Comment:
Why should this throw when the others don't?

Reply:
Because the case has been, in theory, excluded above. So, if this even happens, I would like to know in a very noisy way. Would you prefer to have `assert(false)` here?

Reply:
Yes. It seems like this is not designed to be catchable, so engaging in the "throw/catch" formalism feels like overkill.